src/Controllers/TagController.php

```diff
@@ -14,9 +17,18 @@ class TagController extends LayoutController
      */
     public function showItemByTag(string $tagName = "", int $tagId = null)
     {
+        $category = null;
+        if(RouteConfig::getCategoryId(RouteConfig::SEARCH) > 0)
+        {
+            /** @var CategoryService $categoryService */
+            $categoryService = pluginApp(CategoryService::class);
+            $category = $categoryService->get(RouteConfig::getCategoryId(RouteConfig::SEARCH));
```

Comment:
Im CategoryController wird noch folgende Blöcke ausgeführt, die du in deinem Controller noch ergänzen solltest:

```
$this->categoryService->setCurrentCategory($category);

/** @var ShopBuilderRequest $shopBuilderRequest */
$shopBuilderRequest = pluginApp(ShopBuilderRequest::class);
$shopBuilderRequest->setMainContentType($category->type);
$shopBuilderRequest->setMainCategory($category->id);
```